### ISSUE TITLE
Axis title overlaps with axis labels

### ISSUE THREAD

**Issue by mprudhom** [vega/vega#155]
In the following example spec, the axis labels overlap with the title of the axis:

```javascript
{
  "padding" : "strict",
  "scales" : [
    {
      "name" : "sample.profit",
      "type" : "linear",
      "round" : false,
      "range" : "width",
      "domain" : {
        "field" : "data.profit",
        "data" : "sample"
      }
    },
    {
      "name" : "sample.region",
      "type" : "ordinal",
      "round" : false,
      "range" : "height",
      "domain" : {
        "field" : "data.region",
        "data" : "sample"
      }
    }
  ],
  "marks" : [
    {
      "type" : "rect",
      "key" : "data.region",
      "properties" : {
        "update" : {
          "stroke" : {
            "value" : "transparent"
          },
          "x" : {
            "scale" : "sample.profit",
            "field" : "data.profit"
          },
          "height" : {
            "scale" : "sample.region",
            "mult" : 0.9,
            "band" : true
          },
          "y" : {
            "scale" : "sample.region",
            "field" : "data.region"
          },
          "fillOpacity" : {
            "value" : 1
          },
          "fill" : {
            "value" : "steelblue"
          },
          "strokeWidth" : {
            "value" : 3
          },
          "x2" : {
            "scale" : "sample.profit",
            "value" : 0
          }
        },
        "hover" : {
          "stroke" : {
            "value" : "orange"
          }
        }
      },
      "from" : {
        "data" : "sample"
      }
    }
  ],
  "axes" : [
    {
      "scale" : "sample.profit",
      "type" : "x",
      "title" : "sample.profit"
    },
    {
      "scale" : "sample.region",
      "type" : "y",
      "title" : "sample.region"
    }
  ],
  "data" : [
    {
      "name" : "sample",
      "values" : [
        {
          "region" : "North",
          "profit" : 99
        },
        {
          "region" : "South",
          "profit" : 150
        },
        {
          "region" : "East",
          "profit" : 200
        },
        {
          "region" : "West",
          "profit" : 125
        },
        {
          "region" : "North-East",
          "profit" : 58
        },
        {
          "region" : "North-West",
          "profit" : 188
        },
        {
          "region" : "South-East",
          "profit" : 133
        },
        {
          "region" : "South-West",
          "profit" : 100
        }
      ]
    }
  ]
}
```

**Comment by aramk** [comment]
According to [this answer](http://stackoverflow.com/questions/17797437/d3-autospace-overlapping-tick-labels) d3 doesn't have this either, so perhaps having it there would have the greatest benefit and be utilised by Vega as well?

**Comment by randyzwitch** [comment]
@jheer @arvind  I think this can be closed with release of 2.4?
